### PR TITLE
async-server: fix problems in handle request

### PR DESCRIPTION
**Pull request by Tim-Zhang**
- Return when errors found in handle_request
- Return Code::UNKNOWN(500) when handler failed

### CODE REVIEW

File: src/asynchronous/server.rs [hunk 1]
```diff
@@ -292,6 +292,7 @@ async fn handle_request(
         return;
     }
 
+    let stream_id = header.stream_id;
```
Comment (lifupan) [inline src/asynchronous/server.rs:295]: Why define a new stream_id, why not use header.stream_id directly just as the others did?

Comment (Tim-Zhang) [reply src/asynchronous/server.rs:295]: header.stream_id would be moved when used in line 332

Comment (lifupan) [reply src/asynchronous/server.rs:295]: Please defined it where it needed, don't defined it in the outer scope.

Comment (Tim-Zhang) [reply src/asynchronous/server.rs]: done

**Review by lifupan — APPROVED** [review]

**Review by liubin — APPROVED** [review]

**Pull request merged**